release-notes/3.0/preview/api-diff/preview7/3.0-preview7_System.Collections.Generic.md

```diff
@@ -0,0 +1,15 @@
+# System.Collections.Generic
+
+``` diff
+ namespace System.Collections.Generic {
+-    public interface IDictionary<TKey, TValue> : ICollection<KeyValuePair<TKey, TValue>>, IEnumerable, IEnumerable<KeyValuePair<TKey, TValue>> where TKey : object
++    public interface IDictionary<TKey, TValue> : ICollection<KeyValuePair<TKey, TValue>>, IEnumerable, IEnumerable<KeyValuePair<TKey, TValue>>
+-    public interface IReadOnlyDictionary<TKey, TValue> : IEnumerable, IEnumerable<KeyValuePair<TKey, TValue>>, IReadOnlyCollection<KeyValuePair<TKey, TValue>> where TKey : object
++    public interface IReadOnlyDictionary<TKey, TValue> : IEnumerable, IEnumerable<KeyValuePair<TKey, TValue>>, IReadOnlyCollection<KeyValuePair<TKey, TValue>>
```

Comment:
`where TKey : object` was the previous name for the nullable constraint; now they're `where TKey : notnull`.  It's strange to me to show this as a diff, and more so as a removal of a constraint that's still there, just renamed.

Reply:
I think the reason why this is showing is because the `notnull` constraint  is not emitted into IL and API Reviewer reads this info from metadata.